Consuntivo sprint 4 (#109)
Argo-swe/Docs: sources/documents/2_RTB/PianoDiProgetto/Consuntivo/Sprint/sprint-4.tex

\end{itemize}

\subsubsection{Retrospettiva}

\par Di seguito sono riportati i risultati del questionario di valutazione dello \glossario{sprint}:
\begin{itemize}
  \item Organizzazione dello \glossario{sprint}\ - Valutazione: 7;
  \item Conduzione dei meeting interni - Valutazione: 8;
  \item Conduzione dei meeting esterni - Valutazione: 8;
  \item Impegno e partecipazione dei singoli membri - Valutazione: 9;
  \item La quasi totalità dei membri del team era a conoscenza delle proprie mansioni;
  \item La numerosità delle riunioni è risultata adeguata per circa la metà dei membri, anche se il team preferirebbe organizzare più incontri informali tra programmatori;
  \item Le riunioni sono state organizzate quasi sempre con il giusto preavviso;
  \item Il rapporto ore spese/ore produttive è risultato meno ottimale dello \glossario{sprint} precedente;
  \item La produttività generale ha raggiunto una buona soglia;
  \item Alcuni membri del team ritengono opportuno avere più dettagli anche sulle task meno prossime.
\end{itemize}

\vspace{0.5\baselineskip}
\par A seguire le \textbf{analisi a posteriori} del quarto \glossario{sprint}:
\begin{itemize}

>   \item La sola pianificazione iniziale delle attività non ha permesso ai membri del team di poter svolgere le proprie task a inizio \glossario{sprint}, richiedendo qualche giorno di stallo prima di potersi dedicare alle attività assegnate. Di conseguenza, il gruppo ha ritenuto opportuno, a partire dal prossimo sprint, che i ruoli uscenti spiegassero, tramite riunioni ad hoc, il lavoro che i ruoli entranti dovranno svolgere;

Reviewer: ```suggestion
  \item La sola pianificazione delle attività, per quanto dettagliata, non ha consentito a tutti i membri del team di prendere in carico le proprie task a inizio \glossario{sprint}. Alcuni componenti del gruppo, infatti, hanno incontrato difficoltà nel mantenere l'allineamento con la mole di lavoro elaborata negli sprint precedenti, specialmente riguardo alla coerenza dei contenuti. Per tale motivo il flusso di lavoro ha subito qualche giorno di rallentamento;
\item Il gruppo ha stabilito che, a partire dal prossimo sprint, ciascun componente dovrà documentare, o spiegare tramite riunioni ad hoc, il lavoro pianificato per i ruoli entranti;
```

>   \item Come conseguenza del punto precedente, il team ha stabilito di affidare l'assegnazione delle task per i ruoli entranti ai membri che non ricopreranno più lo stesso ruolo. In questo modo risulterà possibile delineare più puntualmente le attività imminenti;

Reviewer: ```suggestion
  \item Come conseguenza del punto precedente, il team ha deciso di affiancare al responsabile, per l'assegnazione delle task, i membri che hanno già ricoperto un determinato ruolo. In questo modo, il gruppo ritiene di poter delineare con maggior puntualità le attività imminenti;
```

>   \item Anche se i risultati dei questionari di valutazione riguardo le riunioni mostrano una sufficiente adeguatezza organizzativa, il responsabile entrante ha ritenuto opportuno aumentare il margine di preavviso per minimizzare le assenze durante i meeting, comunicando il calendario delle riunioni future entro la fine dello sprint in corso;

Reviewer: ```suggestion
  \item Anche se i risultati del questionario hanno evidenziato una sufficiente adeguatezza organizzativa riguardo le riunioni, il responsabile ha ritenuto opportuno aumentare il margine di preavviso. L'obiettivo è minimizzare le assenze durante i meeting, comunicando il calendario delle riunioni future entro la fine dello sprint;
```

>   \item Nonostante si sia raggiunta una quantità apprezzabile di incontri informali tra membri dai ruoli affini durante la seconda metà dello sprint, lo stesso non si può dire per la prima settimana: questo ha portato ad alcuni periodi di ristagnamento nella prosecuzione delle attività, tenendo comunque alto l'impegno individuale e, di conseguenza, inficiando sul rapporto tra le ore spese e quelle produttive. Preso atto di ciò in vista della riunione di metà sprint, il team ha prontamente affrontato il problema e incrementato il numero di riunioni informative, recuperando così lo stallo iniziale.

Reviewer: ```suggestion
  \item Nonostante il team abbia organizzato una quantità apprezzabile di incontri tra programmatori e altri ruoli correlati durante la seconda metà dello sprint, lo stesso non si può dire per la prima settimana. Ciò ha portato a dei ritardi nella prosecuzione delle attività, mantenendo comunque alto l'impegno individuale e, di conseguenza, inficiando sul rapporto tra le ore spese e quelle produttive. Preso atto di ciò, il gruppo ha prontamente affrontato il problema e ha incrementato il numero di riunioni, recuperando così lo stallo iniziale.
```

\end{itemize}

\subsubsection{Aggiornamento pianificazione e preventivo}
\par Il team ha definito un piano d'azione per migliorare l'organizzazione e la produttività del prossimo \glossario{sprint}:
\begin{itemize}
  \item Aumentare il preavviso per i meeting futuri;

>   \item I ruoli uscenti definiranno le task per i ruoli entranti nel corso dello \glossario{sprint};

Reviewer: ```suggestion
  \item I ruoli uscenti devono affiancare, ove possibile, il responsabile nella definizione delle task per i ruoli entranti;
```

>   \item Incrementare il numero di riunioni informali;

Reviewer: ```suggestion
  \item Incrementare il numero di riunioni informali tra membri che ricoprono ruoli analoghi;
```

>   \item Effettuare una separazione più netta tra \glossario{front-end} e \glossario{back-end};

Reviewer: ```suggestion
  \item Effettuare una separazione più netta tra \glossario{front-end} e \glossario{back-end}.
```

\end{itemize}

\paragraph*{Pianificazione futura:}

\paragraph*{Preventivo "a finire" (\sezione{sec:stima_temporale}):}

> \par Data la numerosità delle risorse assegnate originariamente al ruolo di progettista, il team ha concordato la necessità della ridistribuzione di tali risorse a favore dell'amministratore, il cui impegno orario è stato ritenuto sottostimato, con un lieve aumento anche per i ruoli di programmatore e il verificatore. 

Reviewer: ```suggestion
\par Data la numerosità delle risorse assegnate originariamente al ruolo di progettista, il team ha concordato la necessità di ridistribuire tali risorse a favore dell'amministratore, il cui impegno orario è stato ritenuto sottostimato. Questa ripartizione ha portato a un lieve aumento delle ore produttive totali per i ruoli di programmatore e di verificatore. 
```

La ripartizione è avvenuta come segue:
\begin{itemize}
  \item Le ore complessive assegnate al ruolo di progettista sono diminuite da 161 a 140;
  \item Le ore complessive assegnate al ruolo di amministratore sono aumentate da 56 a 70;
  \item Le ore complessive assegnate al ruolo di programmatore sono aumentate da 154 a 161;
  \item Le ore complessive assegnate al ruolo di verificatore sono aumentate da 140 a 147;
\end{itemize}

> Il monte ore individuali è di conseguenza aumentato da 91 a 92. Nonostante l'incremento delle ore totali per ruolo, la ridistribuzione delle risorse ha comportato un risparmio di €35.

Reviewer: ```suggestion
Il monte ore individuali è di conseguenza aumentato da 91 a 92. Nonostante l'incremento delle ore totali per ruolo, la riallocazione delle risorse ha comportato un risparmio di €35 rispetto al budget previsto per il progetto.
```

\paragraph*{Gestione dei rischi (\sezione{sec:analisi_rischi}):}

> \par Nel corso del quarto \glossario{sprint}, il team ha riscontrato l'affioramento dei seguenti rischi:

Reviewer: ```suggestion
\par Nel corso del quarto \glossario{sprint}, il seguente rischio non è stato gestito con successo:
```

\begin{itemize}

>   \item \textbf{Rischi relativi alla rotazione dei ruoli:} Diversi tra i membri del team hanno riscontrato disorientamento a seguito della rotazione dei ruoli. Una volta individuata tale problematica a seguito di confronti interni tra membri, è stata attuata la contromisura per quanto descritta nella \sezione{sec:analisi_rischi}. L'efficacia del metodo di mitigazione ha subito però una riduzione a causa della sua applicazione tardiva, causando un periodo di stagnamento iniziale.

Reviewer: ```suggestion
  \item \textbf{Rischi relativi alla rotazione dei ruoli:} Diversi membri del team hanno incontrato degli ostacoli a seguito della rotazione dei ruoli. Una volta individuata tale problematica, dopo una serie di confronti interni, è stata attuata la contromisura descritta nella \sezione{sec:analisi_rischi}. L'efficacia del metodo di mitigazione ha subito però una riduzione a causa della sua applicazione tardiva, causando un periodo di rallentamento iniziale. La gestione del rischio deve quindi essere modificata, per aggiungere ulteriori controlli nelle strategie di rilevamento.
```
